ga4gh/frontend.py

```diff
@@ -552,6 +652,65 @@ def searchAlleles(version):
     raise exceptions.NotImplementedException()
 
 
+@app.route('/oauth2callback', methods=['GET'])
+def oidcCallback():
+    """
+    Once the authorization provider has cleared the user, the browser
+    is returned here with a code. This function takes that code and
+    checks it with the authorization provider to prove that it is valid,
+    and get a bit more information about the user (which we don't use).
+
+    A token is generated and given to the user, and the authorization info
+    retrieved above is stored against this token. Later, when a client
+    connects with this token, it is assumed to be a valid user.
+
+    :return: A display of the authentication token to use in the client. If
+    OIDC is not configured, raises a NotImplementedException.
+    """
+    if app.oidcClient is None:
+        raise exceptions.NotImplementedException()
+    response = dict(flask.request.args.iteritems(multi=True))
+    aresp = app.oidcClient.parse_response(message.AuthorizationResponse,
+                                          info=response,
+                                          sformat='dict')
+    sessState = flask.session.get('state')
+    respState = aresp['state']
+    if not isinstance(aresp,
+                      message.AuthorizationResponse) or respState != sessState:
+        raise exceptions.NotAuthenticatedException()
+
+    args = {
+        "code": aresp['code'],
+        "redirect_uri": app.oidcClient.redirect_uris[0],
+        "client_id": app.oidcClient.client_id,
+        "client_secret": app.oidcClient.client_secret
+    }
+    atr = app.oidcClient.do_access_token_request(
+        scope="openid",
+        state=respState,
+        request_args=args)
+
+    if isinstance(atr, message.AccessTokenResponse):
```

Comment:
It would be a little clearer to follow the pattern used above:
```python
if not isinstance(atr, message.AccessTokenResponse):
    raise exceptions.NotAuthenticatedException()
atrDict = atr.to_dict()
# etc.
```